src/components/MessageList/utils/getGroupStyles.ts

```diff
@@ -39,42 +45,46 @@ export const getGroupStyles = <
 >(
   params: GetGroupStylesParams<At, Ch, Co, Ev, Me, Re, Us>,
 ) => {
-  const { messages, noGroupByUser } = params;
+  const {
+    dateSeparators,
+    hideDateSeparators,
+    maxTimeBetweenGroupedMessages,
+    messages,
+    noGroupByUser,
+    userId,
+  } = params;
   const messageGroupStyles: { [key: string]: GroupType[] } = {};
 
-  for (let i = 0; i < messages.length; i++) {
-    const previousMessage = messages[i - 1];
-    const message = messages[i];
-    const nextMessage = messages[i + 1];
-    const groupStyles: GroupType[] = [];
+  const messagesFilteredForNonUser = messages.filter(
+    (message) => !message.deleted_at || userId === message.user?.id,
+  );
 
-    /**
-     * Skip event and date messages
-     */
-    if (message.type === 'channel.event') {
-      continue;
-    }
+  for (let i = 0; i < messagesFilteredForNonUser.length; i++) {
+    const previousMessage = messagesFilteredForNonUser[i - 1];
+    const message = messagesFilteredForNonUser[i];
+    const nextMessage = messagesFilteredForNonUser[i + 1];
+    const groupStyles: GroupType[] = [];
 
     const userId = message?.user?.id || null;
 
     const isTopMessage =
       !previousMessage ||
       previousMessage.type === 'system' ||
-      previousMessage.type === 'channel.event' ||
-      (previousMessage.attachments &&
-        previousMessage.attachments.length !== 0) ||
       userId !== previousMessage?.user?.id ||
       previousMessage.type === 'error' ||
-      !!previousMessage.deleted_at;
+      !!previousMessage.deleted_at ||
+      (!hideDateSeparators && dateSeparators[message.id]);
```

Comment:
```suggestion
      (!hideDateSeparators && dateSeparators[message.id]) ||
      messageGroupStyles[previousMessage.id].indexOf('bottom') > -1;

```

Basically if previous message is `bottom`, then current message should be `top` message!! For our default designs, this extra condition doesn't matter. But if you want to add avatar on top message, this change will be useful.

Reply:
updated to `messageGroupStyles[previousMessage.id]?.includes('bottom');`